### ISSUE TITLE
Editor controls shown when running with `--no-edit`

### ISSUE THREAD

**Issue by geraldcombs** [gollum/gollum#1785]
**Please read the [CONTRIBUTION GUIDELINES](https://github.com/gollum/gollum/blob/master/CONTRIBUTING.md) before submitting!**

If your have problems using or installing the software which stem from bugs in the software or a lack of documentation, we are always happy to help out! However, **for ordinary usage questions, please consider asking elsewhere**, for instance on [StackOverflow](http://stackoverflow.com/questions/tagged/gollum-wiki).

**Describe the bug**

Gollum shows edit controls when running with `--no-edit`.

**To Reproduce**
1. Run `gollum --host localhost --no-edit ...`.
2. open localhost:4567 in your browser.
3. Hover over a heading and click on the pencil icon.
4. Gollum yanks away the football and displays an error page that says "Error / Forbidden. This wiki is set to no-edit mode."

**Expected behavior**

Gollum should probably either hide the pencil icon or show it as grayed / disabled.

**Environment Info**

```
Gollum 5.2.3
Running on: universal.x86_64-darwin21 with Ruby version 2.6.8
Using:
rugged 1.1.1
gollum-rugged_adapter 1.1.2
gollum-lib 5.1.3
Markdown rendering gem: kramdown
Other renderering gems:
none
```

**Comment by bubbasnmp** [comment]
Is it possible to change the text on the error page? That way the user could be pointed to the process needed to update the page using a different route.

**Comment by benjaminwil** [comment]
I think that if you start `gollum` with `--no-edit` nothing is editable from any route.

So I think the only way forward is to either

- override the CSS as suggested previously, and/or
- add some more conditions to the relevant Mustache templates so we can be sure we are not rendering any editing-related HTML/CSS when `--no-edit` is turned on.

**Comment by benjaminwil** [comment]
I would be happy to submit a PR to resolve this issue, but I don't think I have the time to until February.

**Comment by bartkamphorst** [comment]
Thanks @benjaminwil ! I think that timeline is perfectly acceptable. 

> add some more conditions to the relevant Mustache templates

If we can fix it in the templates then that would have my preference.

**Comment by benjaminwil** [comment]
I started investigating this issue more deeply today.

I've discovered a couple of interesting things:

1. `gollum-lib` is responsible for rendering page headings with the class "editable". And `gollum-lib` has no knowledge of the `--no-edit` flag (or the `allow_editing`  setting). [Link to source](https://github.com/gollum/gollum-lib/blob/v5.1.3/lib/gollum-lib/filter/render.rb#L28-L41).
2. If the class "editable" is present on a heading, Gollum adds the edit ✏️ icon link via JavaScript. [Link to source](https://github.com/gollum/gollum/blob/v5.2.3/lib/gollum/public/gollum/javascript/gollum.js.erb#L633-L643)

So I don't think any hacky fixes are required to resolve this issue. It seems it's just a matter of telling `gollum-lib` to _not_ render the "editable" class in some scenarios.

**Comment by benjaminwil** [comment]
I am actually unsure what it means for a heading to have the class "editable". If we removed this class from all headers, what would we be losing? Maybe @dometto knows?

**Comment by benjaminwil** [comment]
I've made a PR against `gollum-lib` as a first step to resolving this issue: https://github.com/gollum/gollum-lib/pull/415

I believe I'm taking a reasonable approach but am open to other strategies.

**Comment by dometto** [comment]
> [](https://github.com/gollum/gollum/issues/1785#issuecomment-1041097927)I am actually unsure what it means for a heading to have the class "editable". If we removed this class from all headers, what would we be losing? Maybe @dometto knows?

The "editable" class indicates that the pencil icon should be shown when hovering above it, and is used for the per-section-editing functionality. Not all headers can be edited per section, as is explained [here](https://github.com/gollum/gollum/wiki/5.0-release-notes#editable-sections). So unfortunately we can't lose the class entirely!

**Comment by benjaminwil** [comment]
I've merged the pull request that resolves this issue: gollum/gollum-lib#415. It should be live in the next release of `gollum-lib`.

**Comment by dometto** [comment]
Released in v5.3.0. (Perhaps we should start labeling issues with `Resolved in next release` or something?).
